packages/client/src/components/block-settings/shared/SelectSettings.tsx

```diff
@@ -0,0 +1,118 @@
+import { useEffect, useMemo, useRef, useState } from 'react';
```

Comment:
Like comment states above if we have a component that already does this, please use that one instead.